recipes/ninja/all/conanfile.py

```diff
@@ -0,0 +1,51 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
+import os
+
+required_conan_version = ">=1.46.0"
+
+
+class NinjaConan(ConanFile):
+    name = "ninja"
+    description = "Ninja is a small build system with a focus on speed"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ninja-build/ninja"
+    topics = ("ninja", "build")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_TESTING"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+        self.conf_info.define("tools.cmake.cmaketoolchain:generator", "Ninja")
```

Comment:
@uilianries @SSE4 @prince-chrismc I believe now that the conan v1 logic `CONAN_CMAKE_GENERATOR` in this recipe shouldn't have been ported to conan v2. It's  not the responsibility of ninja recipe to decide whether CMake generator should be Ninja, even if it's in tool_requires.

Reply:
that's a user's personal preference.

Reply:
agree, so it should be removed

Reply:
Why else would you add it as a requirement? They choose to add it 99% would be to use it... 🤔 

Leave it uless it becomes a problem.

It need more complaints

Reply:
you may add it e.g. for some `meson` dependency. but if it suddenly influences on your `CMake` project by silently changing the generator, it may become a problem. for instance, if library output name depends on generator.
if you add both `ninja` and `make` as build requirements in your profile, they will start fighting for the CMake `generator`.
if it's a personal preference by design, it shouldn't be silently activated from a random dependency in your graph.